src/compiler/checker.ts

```diff
@@ -6170,6 +6174,37 @@ namespace ts {
             }
         }
 
+        function containsArgumentsReference(declaration: FunctionLikeDeclaration): boolean {
+            const links = getNodeLinks(declaration);
+            if (links.containsArgumentsReference === undefined) {
+                if (links.flags & NodeCheckFlags.CaptureArguments) {

```

Comment:
why do we need `traverse` if we have `CaptureArguments`. Is it that not all identifiers are checked in the services case? Or does `CaptureArguments` miss some uses of `arguments`?